bazel/development0.bzl

```diff
@@ -0,0 +1,90 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Load dependencies needed for google-cloud-cpp development / Phase 0."""
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+
+def gl_cpp_development0(name = None):
+    """Loads dependencies needed to develop the google-cloud-cpp libraries.
+
+    google-cloud-cpp developers call this function from the top-level WORKSPACE
+    file to obtain all the necessary *development* dependencies for
+    google-cloud-cpp. This includes testing dependencies and dependencies used
+    by development tools.
+
+    It is a bug if the targets used for google-cloud-cpp development can be used
+    outside the package. All such targets should have their visibility
+    restricted, or are deprecated. If you still need to use such targets, this
+    function may be useful in your own WORKSPACE file.
+
+    This function only loads dependencies that have not been previously loaded,
+    allowing developers to override the version of the dependencies they want to
+    use.
+
+    Args:
+        name: Unused. It is conventional to provide a `name` argument to all
+            workspace functions.
+    """
+
+    # This is needed by com_google_benchmark. We cache it because
+    # sourceforge.net can have outages and that breaks the build.
+    maybe(
+        http_archive,
+        name = "libpfm",
+        urls = [
+            "https://storage.googleapis.com/cloud-cpp-community-archive/libpfm/libpfm-4.11.0.tar.gz",
+            "https://sourceforge.net/projects/perfmon2/files/libpfm4/libpfm-4.11.0.tar.gz",
+        ],
+        sha256 = "5da5f8872bde14b3634c9688d980f68bda28b510268723cc12973eedbab9fecc",
+        strip_prefix = "libpfm-4.11.0",
+        build_file = Label("//bazel:libpfm.BUILD"),
+    )
+
+    # This is only needed to run the microbenchmarks.
+    maybe(
+        http_archive,
+        name = "com_google_benchmark",
+        urls = [
+            "https://storage.googleapis.com/cloud-cpp-community-archive/com_google_benchmark/v1.8.3.tar.gz",
+            "https://github.com/google/benchmark/archive/v1.8.3.tar.gz",
+        ],
+        sha256 = "6bc180a57d23d4d9515519f92b0c83d61b05b5bab188961f36ac7b06b0d9e9ce",
+        strip_prefix = "benchmark-1.8.3",
+    )
+
+    # PugiXML, this is only used in the docfx internal tool.
+    maybe(
+        http_archive,
+        name = "com_github_zeux_pugixml",
+        urls = [
+            "https://storage.googleapis.com/cloud-cpp-community-archive/com_github_zeux_pugixml/v1.14.tar.gz",
+            "https://github.com/zeux/pugixml/archive/v1.14.tar.gz",
+        ],
+        sha256 = "610f98375424b5614754a6f34a491adbddaaec074e9044577d965160ec103d2e",
+        strip_prefix = "pugixml-1.14",
+        build_file = Label("//bazel:pugixml.BUILD"),
+    )
+
+    maybe(
```

Comment:
Nit:  This looks like the only `maybe()` without a short comment.

Reply:
Fixed.